packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -59,9 +166,17 @@ export interface MapProps {
    * The JSON that you want to override your default iteration input
    *
    * @default $
+   * @deprecated use `itemSelector` instead
```

Comment:
Why? We need to also provide an explanation for deprecations. If not for the users, than for the people working on this code base.

Reply:
This is deprecated by StepFunctions service: https://docs.aws.amazon.com/step-functions/latest/dg/input-output-itemselector.html

Would highly recommend going through the complete AWS documentation for the feature to understand how these things work together.

Reply:
Exactly. This explanation should go into the docstring ;)